### ISSUE TITLE
detection of ad is based on player activity only, please we could usage of the client report

### ISSUE THREAD

**Issue by mckaygerhard** [TalkLounge/adblock#3]
it seems you detect the "ad" but in real you only try to guess if player stop suddently to play.. and guess is ad to avoid play to the player.. 

it's not bad but means that the message is not really advertisement about ad.. its just _"i stop to play for a uknow reason but this mod assumed is just ad"_

the best way to made this its by the minetest developer way .. by request  real version to the api at: https://forum.minetest.net/viewtopic.php?p=354631#p354631 

for 0.4 i patched at https://codeberg.org/minenux/minetest-engine-minetest/commit/866f9f78c530b167bb3a3dafb4f61aecf86a4f68
for 5.X i patched at https://codeberg.org/minenux/minetest-engine-minetest/commit/822ab80f30e7982d13b71fc564048c90fca28d7d

**Comment by TalkLounge** [comment]
The client report is built-in since version 5.1.1 as far as i know. So changing the core of the mod would break the backward compatibility to version 0.4.

There is also no guarantee that the client report is correct. Suspicious minetest ad versions are able to just use the official minetest client report since it's completely client side and the server doesn't have a possibility to validate it.

Detect weird movement is still the best way for this mod in my opinion. That's pretty accurate.

If you want to, you can make a pull request, displaying the popup on join when not the official client report is used.

**Comment by mckaygerhard** [comment]
> The client report is built-in since version 5.1.1 as far as i know. So changing the core of the mod would break the backward compatibility to version 0.4.

no .. it's not included never in minetest.. is only active if you build with -dev revision.. 

> There is also no guarantee that the client report is correct. Suspicious minetest ad versions are able to just use the official minetest client report since it's completely client side and the server doesn't have a possibility to validate it.

is always correct.-. as the threath forum demostrates. and i checked its good in my servers, the most used is the 0.4.X protocol

> 
> Detect weird movement is still the best way for this mod in my opinion. That's pretty accurate.
> 

phone clients are suffering in your mod.. i checked you just "guess" the static moment and "could" (assumed) is ad.. is the very bad way to check it.. i was playing in phone and kick me

> If you want to, you can make a pull request, displaying the popup on join when not the official client report is used.

umm ok that's better.. i already do others PR, wait for that ..

**Comment by TalkLounge** [comment]
> no .. it's not included never in minetest.. is only active if you build with -dev revision..

Oh yeah nevermind

> i was playing in phone and kick me

The mod doesn't kick anyone?
Yeah when having really bad internet it can be false positive

**Comment by mckaygerhard** [comment]
> The mod doesn't kick anyone? Yeah when having really bad internet it can be false positive

of course.. also if you stay idle kicked you.. so my aproach its to fech the cllient ifo.. 

if the client are idle BUT also has a not oficial client (using the patch or dev version and by request real version to the api at: https://forum.minetest.net/viewtopic.php?p=354631#p354631 ) then fine.. 

if the patch is not detected just be more flexible and  do not ckick.. just show a formspec but directly with button.. 

so due the second are not so effective.. then we can provide configuration settings as "force_active_users" that will force minimal seconds for idle.. i guess the minimal time to remove ad is:  

`time_show_ad+time_connection+time_click_close+time_reconnection` , by example, popup the ad in phones are almost 1 second, connection in that its almost active , another second to close but phone as slow will take time to respond so another 2 seconds, then reconnect will take almost 6 seconds in best cases..  so minimal are 10 seconds in best conditions..  all of this if the patch or dev version is not detected (remenber that dev version its just flag compilation)
